### PR TITLE
Filter Parameters

### PR DESCRIPTION
**Pull request by nkatzman**
This PR adds in the ability to filter parameters based on indexes.

Relies on: https://github.com/screwdriver-cd/dynamic-dynamodb/pull/4

This will make it easy to scan for values based on indexes. Ensures that we the right limiting is done

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -149,6 +149,7 @@ class Dynamodb extends Datastore {
      * @param  {Object}   config                Configuration object
      * @param  {String}   config.table          Table name
      * @param  {Object}   [config.params]       Record data
+     * @param  {Object}   config.params.equals  Filter data for checking key
```
Comment (stjohnjohnson) [inline index.js]: Why `equals`?  Why not just `params`

Comment (nkatzman) [inline index.js]: What? Nothing to see here..

Comment (nkatzman) [inline index.js]: updated

**Comment by stjohnjohnson** [comment]
LGTM 👍  
bob-o_could_flip_a_burger_like_a_champ__but_when_it_came_to_the_dogs

File: index.js [hunk 2]
```diff
@@ -160,15 +159,23 @@ class Dynamodb extends Datastore {
...
 
         if (!client) {
             const err = new Error(`Invalid table name "${config.table}"`);
 
             return callback(err);
         }
 
-        return client.scan().limit(limitTotalCount)  // scan all items up to and including last item specified
-        .exec((err, data) => {
+        let scanner = client.scan().limit(limitTotalCount);
```
Comment (d2lam) [inline index.js]: Is paginate supposed to be done after or before filtering?

Comment (nkatzman) [inline index.js]: Doesn't really matter. This is just telling the scanner that we want to limit the responses

Comment (nkatzman) [inline index.js]: Ugh just kidding. Spiked it out and if you have `limit()` anywhere, it will only fetch that many records from the db

**Comment by tkyi** [comment]
👍  LGTM
giphy

**Comment by tkyi** [comment]
👍 one more time

**Comment by nkatzman** [comment]
the codacy goes against our eslint rules :/

**Pull request merged**